lib/statsig/logging.ex

```diff
@@ -40,22 +40,30 @@ defmodule Statsig.Logging do
         |> Enum.reduce({[], []}, fn chunk, {failed, successful} ->
           case api_client().push_logs(chunk) do
             {:ok, _} ->
-              {failed, successful ++ chunk}
+              {failed, [successful, chunk]}
 
             {:error, reason} ->
               Logger.error("Failed to flush events: #{inspect(reason)}")
-              {failed ++ chunk, successful}
+              {[failed, chunk], successful}
           end
         end)
 
-      {%{state | events: failed_events}, successful_events}
+      {%{state | events: List.flatten(failed_events)}, List.flatten(successful_events)}
     end
 
     defp flush_interval() do
       Application.get_env(:statsig, :logging_flush_interval, 60_000)
     end
 
     defp api_client, do: Application.get_env(:statsig, :api_client, Statsig.APIClient)
+
+    def schedule_flush(state) do
+      if state.flush_timer != nil do
```

Comment:
nit: the only falsy things are nil and false, so
```suggestion
      if state.flush_timer do
```